src/dbnode/client/session.go

```diff
@@ -1419,6 +1421,13 @@ func (s *session) fetchIDsAttempt(
 	// multiple times in case of retries.
 	ids := inputIDs.Duplicate()
 
+	if s.opts.ProtoEnabled() {
+		schema, schemaErr = s.opts.SchemaRegistry().GetSchema(inputNamespace)
```

Comment:
Could we make all of this logic simpler by just creating an object pool that requires you pass a namespace as an argument? Then there is no possibility of future maintainers introducing bugs in this code path.

```golang
type perNSIterPool struct {
    pool encoding.IteratorPool
    registry namespace.SchemaRegistry
}

func (p *perNSIterPool) Get(ns ident.ID) (encoding.Iterator, error) {
    iter := p.pool.Get()
    schema, err := p.registry.GetSchema(ns)
    if err != nil {
        return err
    }
    iter.SetSchema(schema)
    return iter, nil
}
```

Now no one will ever accidentally do this wrong